templates/shortcode-goal.php

```diff
@@ -36,6 +38,25 @@
 $show_text           = isset( $args['show_text'] ) ? filter_var( $args['show_text'], FILTER_VALIDATE_BOOLEAN ) : true;
 $show_bar            = isset( $args['show_bar'] ) ? filter_var( $args['show_bar'], FILTER_VALIDATE_BOOLEAN ) : true;
 
+/**
+ * @unreleased use DonationQuery to get donation amounts
+ */
+$form_income = 0;
+
+if ($args['start_date'] === $args['end_date']) {
```

Comment:
I might suggest instantiating the query and setting the `->form()` value and then conditionally setting a `between()` value, as opposed to duplicating the `new DonationQuery` code.

On that note, I've considered adding a `->tap($callback)` method to the `DonationQuery` class which would be great for this.

For example:

```php
(new DonationQuery)
    ->form(1)
    ->tap(function($query) {
       if($startDate && $endDate) {
           $query->between($startDate, $endDate);
        }
    })
    ->sumIndendedAmount()
;
```

Reply:
Yeah, I like that.

Reply:
I refactored it a bit https://github.com/impress-org/givewp/pull/7380/commits/7ea042ea03cb5ec63a085cb1a53262ec23dc14d1

If you could add the `tap`  method, that would be great.